Mathlib/Combinatorics/SimpleGraph/CompleteMultipartite.lean

```diff
@@ -0,0 +1,152 @@
+/-
+Copyright (c) 2024 John Talbot and Lian Bremner Tattersall. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: John Talbot, Lian Bremner Tattersall
+-/
+import Mathlib.Combinatorics.SimpleGraph.Coloring
+import Mathlib.Combinatorics.SimpleGraph.Hasse
+
+/-!
+# Complete Multipartite Graphs
+
+A graph is complete multipartite iff non-adjacency is transitive.
+
+## Main declarations
+
+* `SimpleGraph.IsCompleteMultipartite`: predicate for a graph to be complete-multi-partite.
+
+* `SimpleGraph.IsCompleteMultipartite.setoid`: the `Setoid` given by non-adjacency.
+
+* `SimpleGraph.IsCompleteMultipartite.iso`: the graph isomorphism from a graph that
+  `IsCompleteMultipartite` to the corresponding `completeMultipartiteGraph`.
+
+* `SimpleGraph.IsPathGraph3Compl`: predicate for three vertices to be a witness to
+   non-complete-multi-partite-ness of a graph G. (The name refers to the fact that the three
+   vertices form the complement of `pathGraph 3`.)
+-/
+
+universe u
+namespace SimpleGraph
+variable {α : Type u}
+
+/-- `G` is `IsCompleteMultipartite` iff non-adjacency is transitive -/
+def IsCompleteMultipartite (G : SimpleGraph α) : Prop := Transitive (¬ G.Adj · ·)
+
+variable {G : SimpleGraph α}
+/-- The setoid given by non-adjacency -/
+def IsCompleteMultipartite.setoid (h : G.IsCompleteMultipartite) : Setoid α :=
+    ⟨(¬ G.Adj · ·), ⟨G.loopless, fun h' ↦ by rwa [adj_comm] at h', fun h1 h2 ↦ h h1 h2⟩⟩
+
+lemma completeMultipartiteGraph.isCompleteMultipartite {ι : Type*} (V : ι → Type*) :
+    (completeMultipartiteGraph V).IsCompleteMultipartite := by
+  intro
+  aesop
+
+/-- The graph isomorphism from a graph `G` that `IsCompleteMultipartite` to the corresponding
+`completeMultipartiteGraph` -/
```

Comment:
Optional, but I think it could be nice to reference `isCompleteMultipartite_iff` here, since the forward direction of that equivalence seems like a version of this which is more abstract and potentially easier to actually use